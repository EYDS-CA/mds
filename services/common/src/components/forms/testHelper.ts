export const inputMeta = {
    touched: false,
    error: false,
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: false,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: false,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: false,
};

export const errorInputMeta = {
    touched: true,
    error: "Error message",
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: true,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: true,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: true,
}

// needs a value and a name
export const inputProps = {
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onDragStart: jest.fn(),
    onDrop: jest.fn(),
    onFocus: jest.fn()
}