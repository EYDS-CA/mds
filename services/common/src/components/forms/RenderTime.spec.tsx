import React from "react";
import { render } from "@testing-library/react";
import moment from "moment";
import RenderTime from "./RenderTime";
import { inputMeta, inputProps } from "./testHelper";


const time = moment("2019-08-27T07:00:00.000Z");

describe("RenderTime", () => {
    it("renders properly", () => {
        const { container } = render(
            <RenderTime
                id="1"
                label="Time"
                input={{
                    value: time,
                    name: "time-field",
                    ...inputProps
                }}
                fullWidth
                meta={inputMeta}

            />);
        expect(container).toMatchSnapshot();
    });
});
