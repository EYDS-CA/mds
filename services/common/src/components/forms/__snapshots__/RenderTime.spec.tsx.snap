// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderTime renders properly 1`] = `
<div>
  <div
    class="ant-form-item ant-form-item-with-help"
  >
    <div
      class="ant-row ant-form-item-row"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class=""
          for="time-field"
          title="Time"
        >
          Time
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker full"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  autocomplete="off"
                  id="1"
                  name="time-field"
                  placeholder="Select a time"
                  readonly=""
                  size="10"
                  title="00:00"
                  value="00:00"
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="clock-circle"
                    class="anticon anticon-clock-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="clock-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                      />
                      <path
                        d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                      />
                    </svg>
                  </span>
                </span>
                <span
                  class="ant-picker-clear"
                  role="button"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; flex-wrap: nowrap;"
        >
          <div
            class="ant-form-item-explain ant-form-item-explain-connected"
            id="time-field_help"
            role="alert"
          >
            <div
              class=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
